cmd/ocm/gcp/create-wif-config.go

```diff
@@ -0,0 +1,515 @@
+		_, err := client.GetWorkloadIdentityProvider(ctx, providerResource)
+		if err != nil {
+			if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 && strings.Contains(gerr.Message, "Requested entity was not found") {
+				provider := &iam.WorkloadIdentityPoolProvider{
+					Name:        spec.PoolName,
+					DisplayName: spec.PoolName,
+					Description: poolDescription,
+					State:       "ACTIVE",
+					Disabled:    false,
+					Oidc: &iam.Oidc{
+						AllowedAudiences: []string{openShiftAudience},
+						IssuerUri:        spec.IssuerUrl,
+						JwksJson:         spec.Jwks,
+					},
+					AttributeMapping: map[string]string{
+						// when token exchange happens, sub from oidc token shared by operator pod will be mapped to google.subject
+						// field of google auth token. The field is used to allow fine-grained access to gcp service accounts.
+						// The format is `system:serviceaccount:<service_account_namespace>:<service_account_name>`
+						"google.subject": "assertion.sub",
+					},
+				}
+
+				_, err := client.CreateWorkloadIdentityProvider(ctx, fmt.Sprintf("projects/%s/locations/global/workloadIdentityPools/%s", spec.ProjectId, spec.PoolName), spec.PoolName, provider)
+				if err != nil {
+					return errors.Wrapf(err, "failed to create workload identity provider %s", spec.PoolName)
+				}
+				log.Printf("workload identity provider created with name %s", spec.PoolName)
+			} else {
+				return errors.Wrapf(err, "failed to check if there is existing workload identity provider %s in pool %s", spec.PoolName, spec.PoolName)
+			}
+		} else {
+			log.Printf("Workload identity provider %s already exists in pool %s", spec.PoolName, spec.PoolName)
+		}
+
+	}
+	return nil
+}
+
+func createServiceAccounts(ctx context.Context, gcpClient gcp.GcpClient, wifOutput *models.WifConfigOutput, generateOnly bool) error {
+	projectId := wifOutput.Spec.ProjectId
+	fmtRoleResourceId := func(role models.Role) string {
+		return fmt.Sprintf("roles/%s", role.Id)
+	}
+	if generateOnly {
+		for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+			serviceAccountID := serviceAccount.GetId()
+			log.Printf("Would have created service account %s", serviceAccountID)
+			log.Printf("Would have bound roles to %s", serviceAccountID)
+			switch serviceAccount.AccessMethod {
+			case "impersonate":
+				log.Printf("Would have attached impersonator %s to %s", impersonatorServiceAccount, serviceAccountID)
+			case "wif":
+				log.Printf("Would have attached workload identity pool %s to %s", wifOutput.Status.WorkloadIdentityPoolData.IdentityProviderId, serviceAccountID)
+			default:
+				fmt.Printf("Warning: %s is not a supported access type\n", serviceAccount.AccessMethod)
+			}
+		}
+		return nil
+	}
+
+	// Create service accounts
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		serviceAccountID := serviceAccount.GetId()
+		serviceAccountName := wifOutput.Spec.DisplayName + "-" + serviceAccountID
```

Comment:
This is an effective naming scheme, since we will enforce that the display name will be unique for wif_configs in an organization from the backend. We may run into trouble if multiple organizations are making wif configs in the same project, but that is an esoteric edge case.